### PR TITLE
update actions

### PR DESCRIPTION
**Pull request by bangetto**
when I readed back the comments and logs from my previous I saw that some of the actions are still using nodejs 16, so I updated them

### CODE REVIEW

**Review by kennytv — CHANGES_REQUESTED** [review]
Dont set minor/patch versions

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     - name: Checkout Repository
       uses: actions/checkout@v4
     - name: Validate Gradle Wrapper
-      uses: gradle/wrapper-validation-action@v1
+      uses: gradle/wrapper-validation-action@v2.1.1
```
Comment (Kichura) [inline .github/workflows/gradle.yml:13]: Must be `v2` instead of a fixed version.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -21,29 +21,29 @@ jobs:
...
 
       - name: Validate Gradle Wrapper
-        uses: gradle/wrapper-validation-action@v1
+        uses: gradle/wrapper-validation-action@v2.1.1
 
       # Wooyeah
       - name: Read last snapshot
         id: last_snapshot
-        uses: juliangruber/read-file-action@v1
+        uses: juliangruber/read-file-action@v1.1.7
```
Comment (Kichura) [inline .github/workflows/main.yml:36]: Revert this part as my request above applies here along with the same parts on the other two.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -21,29 +21,29 @@ jobs:
...
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: 'temurin'
           java-version: 17
           check-latest: true
 
       - name: Validate Gradle Wrapper
-        uses: gradle/wrapper-validation-action@v1
+        uses: gradle/wrapper-validation-action@v2.1.1
```
Comment (Kichura) [inline .github/workflows/main.yml:31]: Also must be `v2`.

**Review by Kichura — CHANGES_REQUESTED** [review]
Almost LGTM but i noticed some odd parts that need to be resolved first.

**Comment by Kichura** [comment]
Also one more thing; Bump `setup-python` to v5 as that is the latest.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -28,22 +28,22 @@ jobs:
...
       - name: Read last snapshot
         id: last_snapshot
-        uses: juliangruber/read-file-action@v1.1.7
+        uses: juliangruber/read-file-action@v1
         with:
           path: ./last_snapshot.txt
       - name: Read last release
         id: last_release
-        uses: juliangruber/read-file-action@v1.1.7
+        uses: juliangruber/read-file-action@v1.
```
Comment (Kichura) [inline .github/workflows/main.yml:41]: remove the period lol.

**Review by FlorianMichael — APPROVED** [review]
Thanks

**Review by Kichura — APPROVED** [review]

**Review by FlorianMichael — APPROVED** [review]

**Pull request merged**